sys/include/xtimer.h

```diff
@@ -0,0 +1,442 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup  sys_xtimer Timers
+ * @ingroup   sys
+ * @brief     Provides a high level timer module to register
+ *            timers, get current system time, and let a thread sleep for
+ *            a certain amount of time.
+ *
+ * The implementation takes one low-level timer that is supposed to run at 1MHz
+ * speed and multiplexes it.
+ *
+ * Insertion and removal of timers has O(n) complexity with (n) being the
+ * number of active timers.  The reason for this is that multiplexing is
```

Comment:
> 1. cannot break anything that was never there

You're saying you lied in your thesis?

> 1. ..
2. here's no O(1) priority queue, which would be needed to make guarantees unless we limit the number of timers
3. limiting the timers makes it needlessly break for normal usage, with questionable gain in "realtime guarantees"

I was asking for 3. If you don't like, introduce a constant that is 0 or -1 per default as an upper limit and document accordingly.

Seriously, I know that there are several places in RIOT that breaks real-time guarantees, but at least for the core components we should a) try not to introduce new RT-breaking parts (particular not silently) and b) repair the broken stuff.